spec.html

```diff
@@ -28502,24 +28329,18 @@ <h1>Math.fround ( _x_ )</h1>
 
       <emu-clause id="sec-math.hypot">
         <h1>Math.hypot ( _value1_, _value2_, ..._values_ )</h1>
-        <p>`Math.hypot` returns an implementation-approximated value representing the result of the square root of the sum of squares of its arguments.</p>
-        <ul>
-          <li>
-            If no arguments are passed, the result is *+0*.
-          </li>
-          <li>
-            If any argument is *+&infin;*, the result is *+&infin;*.
-          </li>
-          <li>
-            If any argument is *-&infin;*, the result is *+&infin;*.
-          </li>
-          <li>
-            If no argument is *+&infin;* or *-&infin;*, and any argument is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If all arguments are either *+0* or *-0*, the result is *+0*.
-          </li>
-        </ul>
+        <p>Returns the square root of the sum of squares of its arguments.</p>
+        <p>When the `Math.hypot` method is called with at least two arguments _value1_ and _value2_ and any number of additional arguments which form the rest parameter ..._values_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _numbers_ be a List containing _value1_, _value2_, and the elements of _values_ in List order.
+          1. Let _onlyZero_ be *true*.
+          1. For each element _number_ of _numbers_, do
+            1. Let _n_ be ? ToNumber(_number_).
+            1. If _n_ is *NaN*, _n_ is *+&infin;*, or _n_ is *-&infin;*, return _n_.
+            1. If _n_ is neither *+0* nor *-0*, set _onlyZero_ to *false*.
+          1. If _onlyZero_ is *true*, return *+0*.
+          1. Return an implementation-approximated value representing the square root of the sum of squares of the elements of _numbers_.
```

Comment:
Given that the elements of `numbers` aren't actually all numbers - they might be objects that are `ToNumber`ed - it seems like this actually mandates *two* observable ToNumber calls for any object value.

Obviously this is not the intent, and it's likely that all implementations only call ToNumber once per item, but I think we should lock this down in a future PR.

Reply:
Nice catch. I think we should fix that here, probably by constructing a second list while iterating over the first.